tests/unit/query_graph/model/test_column_info.py

```diff
@@ -0,0 +1,185 @@
+"""
+Test ColumnInfo class
+"""
+import pytest
+
+from featurebyte.enum import DBVarType
+from featurebyte.query_graph.model.column_info import ColumnInfo
+from featurebyte.query_graph.model.critical_data_info import CriticalDataInfo
+from featurebyte.query_graph.node.cleaning_operation import (
+    DisguisedValueImputation,
+    MissingValueImputation,
+    UnexpectedValueImputation,
+    ValueBeyondEndpointImputation,
+)
+
+
+@pytest.mark.parametrize(
+    "data_type,imputed_value,expected_imputed_value",
+    [
+        # successful cast
+        (DBVarType.CHAR, "missing", "missing"),
+        (DBVarType.VARCHAR, "missing", "missing"),
+        (DBVarType.VARCHAR, 1234, "1234"),
+        (DBVarType.VARCHAR, True, "True"),
+        (DBVarType.INT, 12, 12),
+        (DBVarType.INT, 12.9, 12),
+        (DBVarType.INT, "12", 12),
+        (DBVarType.FLOAT, 12, 12.0),
+        (DBVarType.FLOAT, 12.9, 12.9),
+        (DBVarType.FLOAT, "12.9", 12.9),
+        (DBVarType.BOOL, 0, False),
+        (DBVarType.BOOL, "", False),
+        (DBVarType.BOOL, 12.9, True),
+        (DBVarType.DATE, "2023-11-09T11:59:15", "2023-11-09"),
+        (DBVarType.TIMESTAMP, "2023-11-09T11:59:15", "2023-11-09T11:59:15"),
+        (DBVarType.TIMESTAMP, "2023-11-09", "2023-11-09T00:00:00"),
+        (DBVarType.TIMESTAMP, "2021-01-01 00:00:00-0800", "2021-01-01T00:00:00"),
```

Comment:
It is expected for the timezone information to be dropped in this case? Seems a bit strange to allow specifying a tz-aware CDI value for a tz-naive column.

Reply:
We have similar case when casting float number to int (where the number after decimal is truncated). Thinking whether it will be a bit inconsistent if we block this case but ok for another case.

Reply:
For timezone we might want to be more careful since dropping timezone information loses much more information than dropping decimals. If the timestamps in the tz-naive column are in UTC, I think a more reasonable casting process is to convert the tz-aware timestamp to UTC first before dropping the timezone.

Reply:
That sounds to be a better solution than failing it. Will update the conversion logic here.